src/ghostfat.c

```diff
@@ -174,8 +175,11 @@ static FileContent_t const info[] = {
     {.name = "CURRENT UF2", .content = NULL       , .size = 0                       },
 };
 
+enum {
+  NUM_FILES = sizeof(info) / sizeof(info[0]),
```

Comment:
Having this in an enum does not prevent bugs from refactoring.  All it does is ensure the result is (essentially) `constexpr`.  If `info` is later refactored to be a pointer, the code breaks.

Using this style of unsafe `ARRAY_SIZE` code is one of the entrenched "bad practices" that I try to help people realize are dangerous.  Please consider using a safe(r) version.

Reply:
thank you, I understand your concern, we don't expose or use this macro on pointer/function argument and is relatively rule out its pit-fall. We can come back to this when we need an global ARRAY_SIZE, or we could come back to this later on. 

PS: I take a quick look at simple hack, it is great hack. Though I think we could simplify its array size more, will do it in another PR later on.